test/functional/lua/version_spec.lua

```diff
@@ -1,186 +1,150 @@
-        v2 = '1.0.0-alpha',
-        want = 1,
-      },
-      {
-        desc = '(v1 < v2) when v2 has an additional alphabet identifier',
-        v1 = '1.0.0-alpha',
-        v2 = '1.0.0-alpha.beta',
-        want = -1,
-      },
-      {
-        desc = '(v1 < v2) when v2 has an a first alphabet identifier with higher precedence',
-        v1 = '1.0.0-alpha.beta',
-        v2 = '1.0.0-beta',
-        want = -1,
-      },
-      {
-        desc = '(v1 > v2) when v1 has an a first alphabet identifier with higher precedence',
-        v1 = '1.0.0-beta',
-        v2 = '1.0.0-alpha.beta',
-        want = 1,
-      },
-      {
-        desc = '(v1 < v2) when v2 has an additional number identifer',
-        v1 = '1.0.0-beta',
-        v2 = '1.0.0-beta.2',
-        want = -1,
-      },
-      {
-        desc = '(v1 < v2) when v2 has same first alphabet identifier but has a higher number identifer',
-        v1 = '1.0.0-beta.2',
-        v2 = '1.0.0-beta.11',
-        want = -1,
-      },
-      {
-        desc = '(v1 < v2) when v2 has higher alphabet precedence',
-        v1 = '1.0.0-beta.11',
-        v2 = '1.0.0-rc.1',
-        want = -1,
-      },
+      { v1 = 'v0.0.99',          v2 = 'v9.0.0',           want = -1, },
+      { v1 = 'v0.4.0',           v2 = 'v0.9.99',          want = -1, },
+      { v1 = 'v0.2.8',           v2 = 'v1.0.9',           want = -1, },
+      { v1 = 'v0.0.0',           v2 = 'v0.0.0',           want = 0, },
+      { v1 = 'v9.0.0',           v2 = 'v0.9.0',           want = 1, },
+      { v1 = 'v0.9.0',           v2 = 'v0.0.0',           want = 1, },
+      { v1 = 'v0.0.9',           v2 = 'v0.0.0',           want = 1, },
+      { v1 = 'v1.0.0-alpha',     v2 = 'v1.0.0',           want = -1, },
+      { v1 = '1.0.0',            v2 = '1.0.0-alpha',      want = 1, },
+      { v1 = '1.0.0-2',          v2 = '1.0.0-1',          want = 1, },
+      { v1 = '1.0.0-2',          v2 = '1.0.0-9',          want = -1, },
+      { v1 = '1.0.0-2',          v2 = '1.0.0-2.0',        want = -1, },
+      { v1 = '1.0.0-2.0',        v2 = '1.0.0-2',          want = 1, },
+      { v1 = '1.0.0-2.0',        v2 = '1.0.0-2.0',        want = 0, },
+      { v1 = '1.0.0-alpha',      v2 = '1.0.0-alpha',      want = 0, },
+      { v1 = '1.0.0-alpha',      v2 = '1.0.0-beta',       want = -1, },
+      { v1 = '1.0.0-beta',       v2 = '1.0.0-alpha',      want = 1, },
+      { v1 = '1.0.0-alpha',      v2 = '1.0.0-alpha.1',    want = -1, },
+      { v1 = '1.0.0-alpha.1',    v2 = '1.0.0-alpha',      want = 1, },
+      { v1 = '1.0.0-alpha.beta', v2 = '1.0.0-alpha',      want = 1, },
+      { v1 = '1.0.0-alpha',      v2 = '1.0.0-alpha.beta', want = -1, },
+      { v1 = '1.0.0-alpha.beta', v2 = '1.0.0-beta',       want = -1, },
+      { v1 = '1.0.0-beta',       v2 = '1.0.0-alpha.beta', want = 1, },
+      { v1 = '1.0.0-beta',       v2 = '1.0.0-beta.2',     want = -1, },
+      -- TODO
+      -- { v1 = '1.0.0-beta.2',     v2 = '1.0.0-beta.11',    want = -1, },
```

Comment:
@folke IIUC https://semver.org/ , this should be: 

    1.0.0-beta.11 > 1.0.0-beta.2

but I think lazy currently treats this as `<` ?

Reply:
oh you're right. Seems I didn't implement that part [.4](https://semver.org/#spec-item-11)

Reply:
I remember now. For lazy's use-case that doesn't really matter, since if you want to get the latest version, you always get the latest stable without including pre-releases.

In Semver ranges, you can only use pre-release versions when they are explicitly specified.